compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp

```diff
@@ -0,0 +1,222 @@
+  b.setInsertionPointToStart(&topLevelTransformRegion.front());
+  auto anyOpType = transform::AnyOpType::get(b.getContext());
+
+  // Create the include for the named sequence with the expectation that the
+  // external definition will be linked in later.
+  auto sequence = b.create<transform::SequenceOp>(
+      loc, TypeRange{}, transform::FailurePropagationMode::Propagate, anyOpType,
+      [&](OpBuilder &b, Location loc, Value variantH) {
+        b.create<transform::PrintOp>(loc, variantH);
+        b.create<transform::YieldOp>(loc);
+      });
+  (void)sequence;
+}
+
+struct MaterializeUserConfigsPass
+    : public MaterializeUserConfigsBase<MaterializeUserConfigsPass> {
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registerTransformDialectTranslationDependentDialects(registry);
+  }
+
+  void runOnOperation() override {
+    IREE::HAL::ExecutableVariantOp variantOp = getOperation();
+    ModuleOp moduleOp = variantOp.getInnerModule();
+    llvm::StringMap<IREE::HAL::ExecutableExportOp> exportOps =
+        getAllEntryPoints(moduleOp);
+    MLIRContext *context = moduleOp.getContext();
+
+    std::optional<ModuleOp> transformLibrary = std::nullopt;
+    if (!clCodegenTransformDialectLibraryFileName.empty()) {
+      auto dialect =
+          context->getOrLoadDialect<IREE::Codegen::IREECodegenDialect>();
+      auto maybeTransformLibrary = dialect->getOrLoadTransformLibraryModule(
+          clCodegenTransformDialectLibraryFileName);
+      if (failed(maybeTransformLibrary)) {
+        return signalPassFailure();
+      }
+      transformLibrary = *maybeTransformLibrary;
+    }
+
+    IREE::Codegen::DispatchLoweringPassPipeline tdPipeline =
+        IREE::Codegen::DispatchLoweringPassPipeline::TransformDialectCodegen;
+    std::optional<IREE::Codegen::TranslationInfoAttr> clTranslationInfo;
+    // Here we always set the pipeline strategy to transform dialect if the
+    // flag is non-empty to ensure we pick the right lowering pipeline in the
+    // event a file path is given.
+    if (!clCodegenTransformDialectTestName.empty()) {
+      clTranslationInfo = IREE::Codegen::TranslationInfoAttr::get(
+          context, tdPipeline,
+          /*softwarePipelineDepth=*/0,
+          /*softwarePipelineStoreStage=*/1,
+          /*codegenSpec=*/clCodegenTransformDialectTestName[0] == '@'
+              ? SymbolRefAttr::get(
+                    context, llvm::StringRef(
+                                 clCodegenTransformDialectTestName.substr(1)))
+              : SymbolRefAttr());
+    }
+
+    std::optional<IREE::Codegen::TranslationInfoAttr> translationInfo;
+    for (auto funcOp : moduleOp.getOps<func::FuncOp>()) {
+      auto exportOp = exportOps.lookup(funcOp.getName());
+      if (!exportOp) {
+        continue;
+      }
+
+      /// First, apply all user configs.
+      auto res = funcOp.walk([&](Operation *op) {
+        if (auto compilationInfo = getCompilationInfo(op)) {
```

Comment:
This logic is repeated here https://github.com/openxla/iree/pull/14788/files#diff-1ac727f2f0f42c0b7710d5c4778ec0b83e5a318765d2eb6d5960602a03f76f5dL2396

here https://github.com/openxla/iree/pull/14788/files#diff-e491707653524e61bd22ba47ca5bd8414065b70bd1c6b85f45947929d7a49aafL1175

Maybe those could be removed now.

Reply:
(for whatever reason the links aren't opening to the right place for me, but I assume you're talking about the other places where setUserConfig is called, e.g. https://github.com/openxla/iree/blob/446f96eb896dcc5a1d29ff0af3363d8df1fe7021/compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp#L2399)

I left those alone because I figured it was still convenient to attach a user config to a dispatch and run kernel-config without the need to run this Materialize pass first. I'm fine with removing it though.

Reply:
yeah, I am not sure we do that really, and maybe just better consolidate all the user configuration setting in one place.

Reply:
> (for whatever reason the links aren't opening to the right place for me, but I assume you're talking about the other places where setUserConfig is called, e.g.
> 

ugh... not sure why the links didnt open, maybe I shouldnt have sent the links in the review page...